website/pages/chapters/ridge.js

```diff
@@ -0,0 +1,845 @@
+          Unfortunately, this will not work! Consider the case where one
+          coefficient is a number like <IM math={`10,002`} /> and another is{" "}
+          <IM math={`-10,001`} />. Even though they each have very large
+          magnitudes, if you add them together, they cancel each other out
+          resulting in a sum of <IM math={`-1`} />. This doesn't quite capture
+          our intent of making <IM math={`R(w)`} /> large if the magnitudes of
+          the coefficients are large.
+        </p>
+
+        <p>
+          While the last idea didn't work, it's actually quite close to one that
+          does! Instead of taking the sum of the coefficients, we can take the
+          sum of their absolute values to remove the negations.
+        </p>
+
+        <BM math={`R(w) = \\sum_{j=0}^D |w_j|`} />
+
+        <p>
+          Mathematicians call this value the <b>L1 norm</b> and use the
+          short-hand notation <IM math={`||w||_1`} /> when using it. In other
+          words, we are using <IM math={`R(w) = ||w||_1`} /> as our measure of
+          overfitting. You can hopefully see why this solves the problem we had
+          before. The L1 norm will be larger as the coefficients grow in
+          magnitude and the coefficients can't cancel out since we take that
+          absolute value.
+        </p>
+
+        <p>
+          We won't actually discuss this L1 norm in detail now, but will reserve
+          discussion for it in the next chapter. For this chapter, we will use
+          another common norm as our measure of magnitude called the{" "}
+          <b>L2 norm</b>.
+        </p>
+
+        <BM math={`R(w) = \\sum_{j=0}^D w_j^2`} />
+
+        <p>
+          We use the short-hand notation <IM math={`||w||_2^2`} /> to describe
+          the L2 norm
+          <MarginNote>
+            📝 <em>Notation:</em> The L1 and L2 norms are specific examples of a{" "}
+            <em>p</em>-norm. The <em>p</em>-norm is defined for any positive
+            number <IM math={`p`} /> and has the value{" "}
+            <IM
+              math={`||w||_p = \\left(\\sum_{j=0}^D\\left|w_j\\right|^p\\right)^{\\frac{1}{p}}`}
+            />
+            . To avoid dealing with square roots with the L2 norm, we square it.
+            This why we use the L2 norm notation <IM math={`||w||_2^2`} /> is
+            squared.
+          </MarginNote>
+          . In other words, we are using <IM math={`R(w) = ||w||_2^2`} />. You
+          can convince yourself that this will also solve the cancelling-out
+          problem like the L1 norm. In the next chapter, we will see how the
+          differing effects of the L1 and L2 norms, but for the rest of this
+          chapter, we will focus on this L2 norm.
+        </p>
+
+        <h3>Ridge Regression</h3>
+        <p>
+          When we use regularization in our regression context with the L2 norm,
+          our quality metric then becomes the <b>ridge regression</b> quality
```

Comment:
do we have any intuition for why it's called ridge? might be helpful to associate this with the l2 norm. I don't remember any, it only took me 3 ml classes to commit ridge and lasso to memory lol.

Reply:
From Wikipedia: 

>  In statistics, the method is known as ridge regression, in machine learning it is known as weight decay, and with multiple independent discoveries, it is also variously known as the Tikhonov–Miller method, the Phillips–Twomey method, the constrained linear inversion method, and the method of linear regularization. It is related to the Levenberg–Marquardt algorithm for non-linear least-squares problems.

Reply:
Ah it looks like it is trying to fix the "ridge" problem where you have a co-linearity in your data and there is no unique global minimum to the RSS. 

![image](https://user-images.githubusercontent.com/9054538/89698148-74a72980-d8d4-11ea-8f6c-c3a90aafb38e.png)